### ISSUE TITLE
Consider requiring task id as part of Activity scope names

### ISSUE THREAD

**Issue by ghost** [square/mortar#48]
We ran into an issue where if you really whale on the app icon and the home button you can see a new task start and replace the current one. Activities in the outgoing task will have their onDestroy() method called only after their peers in the new task get created. Then you crash because the old activity has ended the scope that the new activity is trying to inject from. 

The fix is simple, in our base Activity class:

``` java
  protected Blueprint getMortarBlueprint() {
    return new Blueprint() {
      @Override public String getMortarScopeName() {
        return SquareActivity.this.getMortarScopeName();
      }

      @Override public Object getDaggerModule() {
        return getModules();
      }
    };
  }

  protected String getMortarScopeName() {
    // If you really whale on the app icon and the home button you can see
    // a new task start and replace the current one. Activities in the
    // outgoing task will have their onDestroy() method called only after
    // their peers in the new task get created. Include the task id in the scope
    // name to stop them stomping each other. RA-1760
    return getClass().getName() + "-task-" + getTaskId();
  }
```

I'm not certain yet if this is specific to our use of activity config and the back stack, etc. I suspect not. 

We should consider making requireActivityScope require the taskId and wrap the Blueprint so that it appends the taskId to the given name.
